packages/default/scss/common/_forms.scss

```diff
@@ -263,15 +263,8 @@ $input-border-radius: $border-radius !default;
     .ng-invalid.ng-dirty {
         &.k-autocomplete,
         &.k-maskedtextbox > .k-textbox {
-
-            &,
-            &.k-state-hover,
-            &:hover,
-            &.k-state-focused,
-            &:focus {
-                color: $error;
-                border-color: $invalid-border;
-            }
```

Comment:
After those changes, the red invalid border is changed when a widget is hovered/selected. It should be always red.